misc/path_utils.c

```diff
@@ -151,10 +151,61 @@ char *mp_getcwd(void *talloc_ctx)
 
 char *mp_normalize_path(void *talloc_ctx, const char *path)
 {
+    if (!path)
+        return NULL;
+
     if (mp_is_url(bstr0(path)))
         return talloc_strdup(talloc_ctx, path);
 
-    return mp_path_join(talloc_ctx, mp_getcwd(talloc_ctx), path);
+    if (!mp_path_is_absolute(bstr0(path))) {
+        char *cwd = mp_getcwd(talloc_ctx);
+        if (!cwd)
+            return NULL;
+        path = mp_path_join(talloc_ctx, cwd, path);
+    }
+
+#if HAVE_DOS_PATHS
+    return talloc_strdup(talloc_ctx, path);
```

Comment:
For Windows, probably this should be used https://learn.microsoft.com/en-us/windows/win32/api/pathcch/nf-pathcch-pathcchcanonicalizeex

Reply:
Can you implement calling that after merging this? I have no idea how to write Windows code.

Reply:
done